notifier/notifier.go

```diff
@@ -383,25 +391,25 @@ func (n *Notifier) Stop() {
 
 // Describe implements prometheus.Collector.
 func (n *Notifier) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
I'm super-interested in this because I want to solve this "the one true way" in client_golang 0.10.

@juliusv Do I understand correctly that you want joined metrics for _all_ instances of the AM chain? I.e. without an additional label that would separate out each instance in the metrics (like the user ID owning that AM chain). In that case, I believe, you _have_ to share the metrics. If you do have a partitioning label, you could use it as const-label and it would register with the same Registerer just fine, despite the same name.

In very general, I think the global pattern is neat for non-libraries of limited size. But as @fabxc says, how do you know you are a non-library? Anything global is so frowned upon in the Go community that there is no way it can become the recommended way. (It should certainly stay possible.) For the non-global way, the competing models so far are implementing `Collector` and register individual instances with a `Registerer` of your choice after construction, or passing in a `Registerer` so that the constructor registers metrics itself. The latter has only become possible recently, and it might be the overall more elegant way. Neither, though, seems to support @juliusv 's use-case.

Anyway, don't want to derail the discussion at hand. Just super-keen to see which pattern "wins" and to support it in upcoming improvements of client_golang.